apps/volunteer/sign_up.py

```diff
@@ -82,8 +70,12 @@ def update_volunteer_from_form(volunteer, form):
 @feature_flag("VOLUNTEERS_SIGNUP")
 def sign_up():
     form = VolunteerSignUpForm()
-    # On sign up give user 'volunteer' permission (+ managers etc.)
+    form.arrival.choices = generate_arrival_options()
+    form.arrival.data = event_start().strftime("%F")
+    form.departure.choices = generate_departure_options()
+    form.departure.data = event_end().strftime("%F")
```

Comment:
I'm a little new to the `data` attribute for these form fields - could you explain what these assignments are for and how they affect the sign-up experience?

Reply:
`data` sets the value of the Select on display so in this case the first and last days of the event will be the defaults presented.

We're using to display the data elsewhere in the same file. The docs are not great for `SelectField`

Reply:
Ok, makes sense - that display logic makes sense when `sign_up` is called as a `GET`.  It  looks like `sign_up` also handles form content submission via `POST`.

Could you test (either locally or, preferably by adding some test coverage) that the volunteer arrival & departure dates are populated on form submission?

Reply:
Manually tested and it looks good. Other than populating the choices before the form is rendered that shouldn't have affected anything.